tests/components/huawei_lte/test_config_flow.py

```diff
@@ -183,21 +259,93 @@ async def test_ssdp(hass):
             ssdp_location="http://192.168.100.1:60957/rootDesc.xml",
             upnp={
                 ssdp.ATTR_UPNP_DEVICE_TYPE: "urn:schemas-upnp-org:device:InternetGatewayDevice:1",
-                ssdp.ATTR_UPNP_FRIENDLY_NAME: "Mobile Wi-Fi",
                 ssdp.ATTR_UPNP_MANUFACTURER: "Huawei",
                 ssdp.ATTR_UPNP_MANUFACTURER_URL: "http://www.huawei.com/",
                 ssdp.ATTR_UPNP_MODEL_NAME: "Huawei router",
                 ssdp.ATTR_UPNP_MODEL_NUMBER: "12345678",
                 ssdp.ATTR_UPNP_PRESENTATION_URL: url,
-                ssdp.ATTR_UPNP_SERIAL: "00000000",
                 ssdp.ATTR_UPNP_UDN: "uuid:XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+                **upnp_data,
             },
         ),
     )
 
+    for k, v in expected_result.items():
+        assert result[k] == v
+    if result.get("data_schema"):
+        result["data_schema"]({})[CONF_URL] == url
+
+
+@pytest.mark.parametrize(
+    ("login_response_text", "expected_result", "expected_entry_data"),
+    (
+        (
+            "<response>OK</response>",
+            {
+                "type": data_entry_flow.FlowResultType.ABORT,
+                "reason": "reauth_successful",
+            },
+            FIXTURE_USER_INPUT,
+        ),
+        (
+            f"<error><code>{LoginErrorEnum.PASSWORD_WRONG}</code><message/></error>",
+            {
+                "type": data_entry_flow.FlowResultType.FORM,
+                "errors": {CONF_PASSWORD: "incorrect_password"},
+                "step_id": "reauth_confirm",
+            },
+            {**FIXTURE_USER_INPUT, CONF_PASSWORD: "invalid-password"},
+        ),
+    ),
+)
+async def test_reauth(
+    hass, login_requests_mock, login_response_text, expected_result, expected_entry_data
+):
+    """Test reauth."""
+    mock_entry_data = {**FIXTURE_USER_INPUT, CONF_PASSWORD: "invalid-password"}
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=FIXTURE_UNIQUE_ID,
+        data=mock_entry_data,
+        title="Reauth canary",
+    )
+    entry.add_to_hass(hass)
+
+    context = {
+        "source": config_entries.SOURCE_REAUTH,
+        "unique_id": entry.unique_id,
+        "entry_id": entry.entry_id,
+    }
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context=context, data=entry.data
+    )
+
     assert result["type"] == data_entry_flow.FlowResultType.FORM
-    assert result["step_id"] == "user"
-    assert result["data_schema"]({})[CONF_URL] == url
+    assert result["step_id"] == "reauth_confirm"
+    assert result["data_schema"]({}) == {
+        CONF_USERNAME: mock_entry_data[CONF_USERNAME],
+        CONF_PASSWORD: mock_entry_data[CONF_PASSWORD],
+    }
+    assert not result["errors"]
+
+    login_requests_mock.request(
+        ANY,
+        f"{FIXTURE_USER_INPUT[CONF_URL]}api/user/login",
+        text=login_response_text,
+    )
+    result = await hass.config_entries.flow.async_configure(
```

Comment:
We normally patch the integration setup function in the reauth flow tests on success to avoid setting up the integration when the config entry is reloaded.